pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -286,12 +287,15 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication, VaultGuard {
         for (uint256 i = 0; i < poolTokens.length; ++i) {
             IERC20 token = poolTokens[i];
 
-            (totalSwapFees[i], totalYieldFees[i]) = _aggregateFeeAmounts[pool][token].fromPackedBalance();
+            // If a feeToken is specified, all reported fees will be zero except for that token.
+            if (feeToken == IERC20(address(0)) || feeToken == token) {
```

Comment:
maybe do a boolean 
```
bool collectAllTokens = (feeToken == IERC20(address(0)));
```

outside the outer for loop and use it here for just a bit more clarity?